libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -50,4 +53,92 @@ impl ScabbardClient {
             Ok(())
         }
     }
+
+    pub fn get_state_at_address(
+        &self,
+        circuit_id: &str,
+        service_id: &str,
```

Comment:
If we create a struct "ServiceId" here, it could include both circuit_id and service_id. That struct could enforce correctness (validate that they appear to be valid from a length and character set perspective), making it safer input into these functions.

Reply:
Add the `ServiceId` struct.